src/Rules/Data.hs

```diff
@@ -29,6 +29,13 @@ buildPackageData context@Context {..} = do
         -- GhcCabal may run the configure script, so we depend on it.
         whenM (doesFileExist $ configure <.> "ac") $ need [configure]
 
+        -- pkgs not requiring inplaceLibCopyTargets to build
+        ghcCabalDeps <- pkgDependencies ghcCabal
+        deriveConstantsDeps <- pkgDependencies ghcCabal
```

Comment:
Yes! Sorry for the mistake, fixed now. I guess it is because `deriveConstants` and `ghcCabal` has overlapped dependencies so it is working fine.